code/Player.cs

```diff
@@ -1,23 +1,40 @@
 ﻿using Sandbox;
+using TerryForm.Weapons;
 
 namespace TerryForm
 {
-	public class Player : Sandbox.Player
+	public partial class Player : Sandbox.Player
 	{
+		[Net] public Weapon EquippedWeapon { get; set; }
+
 		public override void Respawn()
 		{
+			SetModel( "models/maya_testcube_100.vmdl" );
+
 			// Need a custom controller!!
 			Controller = new WalkController();
 			Camera = new Camera();
 
+			EquipWeapon( new Weapon() );
+
 			base.Respawn();
 		}
 
-		public override void Simulate( Client cl )
+		protected void EquipWeapon( Weapon weapon )
 		{
-			base.Simulate( cl );
+			EquippedWeapon?.Delete();
+
+			EquippedWeapon = weapon;
+			EquippedWeapon?.OnCarryStart( this );
+		}
 
+		public override void Simulate( Client cl )
+		{
 			DebugOverlay.Sphere( EyePos, 32f, Color.Yellow, false ); // Visualise the pawn since we don't have a model for it yet
+
+			SimulateActiveChild( cl, EquippedWeapon ); //Simulate our currently equipped weapon.
```

Comment:
Comments should be prefixed with a padding space (see line above) but, as Trundler mentioned on Discord, inline comments are probably bad